pkg/controller/mount_controller.go

```diff
@@ -102,6 +111,126 @@ func (m MountController) Reconcile(ctx context.Context, request reconcile.Reques
 	return reconcile.Result{}, err
 }
 
+// handlePendingMountPod handles pending mount pod that is not scheduled yet (nodeName is empty)
+// It checks if all referenced app pods are deleted, and if so, deletes the mount pod
+func (m *MountController) handlePendingMountPod(ctx context.Context, mountPod *corev1.Pod) (reconcile.Result, error) {
+	mountCtrlLog.V(1).Info("Handling pending mount pod", "name", mountPod.Name, "namespace", mountPod.Namespace)
+
+	// Get target node name from mount pod's nodeSelector
+	// This helps optimize pod search by limiting to a specific node
+	targetNodeName := ""
+	if mountPod.Spec.NodeSelector != nil {
+		targetNodeName = mountPod.Spec.NodeSelector["kubernetes.io/hostname"]
+	}
+
+	// Check all references in mount pod annotations
+	var existingRefs int
+	for k, target := range mountPod.Annotations {
+		if k == util.GetReferenceKey(target) {
+			// Extract pod UID from target path
+			// Target format: /var/lib/kubelet/pods/<pod-uid>/volumes/kubernetes.io~csi/<volume-id>/mount
+			targetUid := getPodUid(target)
+			if targetUid == "" {
+				mountCtrlLog.V(1).Info("Could not extract pod UID from target", "target", target)
+				continue
+			}
+
+			// Find pod by UID, optimized by searching only on target node
+			targetPod, err := m.GetPodByUidAndNode(ctx, targetUid, targetNodeName)
+			if err != nil {
+				mountCtrlLog.Error(err, "Failed to search for app pod by UID", "appPodUid", targetUid)
+				return reconcile.Result{}, err
+			}
+
+			if targetPod != nil {
+				mountCtrlLog.V(1).Info("Referenced app pod still exists",
+					"appPod", targetPod.Name,
+					"appPodNamespace", targetPod.Namespace,
+					"appPodUid", targetUid,
+					"mountPod", mountPod.Name)
+				existingRefs++
+			} else {
+				mountCtrlLog.Info("Referenced app pod has been deleted",
+					"appPodUid", targetUid,
+					"mountPod", mountPod.Name)
+			}
+		}
+	}
+
+	// If no app pods reference this mount pod anymore, delete it
+	if existingRefs == 0 {
+		// Check if should delay deletion
+		shouldDelay, err := resource.ShouldDelay(ctx, mountPod, m.K8sClient)
```

Comment:
pending pod does not need to consider delay

Reply:
delete